### ISSUE TITLE
Crash on Android 7.1

### ISSUE THREAD

**Issue by simonspa** [stefan-niedermann/nextcloud-deck#458]
**Describe the bug**
Got this stack trace when starting:

```
App Version: 1.0.2
App Version Code: 1000002
App Flavor: play

Files App Version Code: 30110190

---
OS Version: 3.4.0+(e730b2f661)
OS API Level: 25
Device: onyx
Manufacturer: OnePlus
Model (and Product): ONE E1003 (onyx)

---

java.lang.RuntimeException: Unable to start activity ComponentInfo{it.niedermann.nextcloud.deck.play/it.niedermann.nextcloud.deck.ui.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Long it.niedermann.nextcloud.deck.model.Board.getLocalId()' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2684)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6186)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Long it.niedermann.nextcloud.deck.model.Board.getLocalId()' on a null object reference
	at it.niedermann.nextcloud.deck.ui.MainViewModel.getCurrentBoardLocalId(MainViewModel.java:43)
	at it.niedermann.nextcloud.deck.ui.stack.StackFragment.onCreateView(StackFragment.java:67)
	at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2698)
	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:320)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1187)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
	at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1497)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2625)
	at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2577)
	at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:247)
	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:541)
	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:201)
	at it.niedermann.nextcloud.deck.ui.branding.BrandedActivity.onStart(BrandedActivity.java:53)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1249)
	at android.app.Activity.performStart(Activity.java:6701)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2647)
	... 9 more
```

**To Reproduce**
1. Start the app
2. See error

**Expected behavior**
:-)

**Comment by stefan-niedermann** [comment]
Hmm... a few questions:

- did it work once and stopped working or did you start the app the very first time?
- have you done an update or is this version a fresh install?
- have you recently deleted a board?

Could you please try to delete the storage of the app and readd your account? (android settings -> apps -> Nextcloud Deck -> storage -> delete storage)

**Comment by simonspa** [comment]
It's a bit mysterious indeed:

* mostly it works just fine, the crash only appears spuriously
* normally, the app works flawlessly
* this is a fresh install of 1.0

One suspicion I have is that it has to do with network changes: I currently test Deck and this app with my testing NC server only reachable from within my LAN. It "feels" like the crashes appear after I have been outside the network. I'll try to narrow this down...

**Comment by stefan-niedermann** [comment]
I have also an idea and will further investigate it.

It might get caused by applying the branding before the last used board could be read from storage. So if you are able to reproduce the issue on a regular base you could disable the branding in the app settings to verify if this prevents the crashes.

**Comment by stefan-niedermann** [comment]
Have you been able to reproduce this issue? Could you verify your suspicion, that it has something to do with network changes?

**Comment by simonspa** [comment]
I played around but I can't reproduce it really. It sometimes reappeared, mostly after the app was dormant for a while and I reopen it. Unfortunately nothing more concrete, sorry... :/

**Comment by stefan-niedermann** [comment]
I found the cause of the issue. It appears, when you put the app into background, the Android system itself kills the app (for example because of low resources) and you switch back to the app. 

That's why it is so hard to reproduce - Android decides *when* it needs resources.

We will track and fix this in #478 where i described the complete scenario.

Thank you for your report and help.
